packages/ui/src/hooks/useHover.ts

```diff
@@ -0,0 +1,40 @@
+import React, {useEffect, useRef, useState} from 'react';
+
+type UseHoverType<T extends HTMLElement> = [React.RefObject<T>, boolean];
+
+export function useHover<T extends HTMLElement>(
+  handleMouseOver?: () => void,
+  handleMouseOut?: () => void,
+): UseHoverType<T> {
+  const [value, setValue] = useState(false);
+
+  const ref = useRef<T>(null);
+
+  const _handleMouseOver = () => {
+    setValue(true);
+    handleMouseOver?.();
+  };
+
+  const _handleMouseOut = () => {
+    setValue(false);
+    handleMouseOut?.();
+  };
+
+  useEffect(
+    () => {
+      const node = ref.current;
+      if (node) {
+        node.addEventListener('mouseover', _handleMouseOver);
```

Comment:
Is `mouseover` necessary? Seems like `mouseenter` would be enough.

Reply:
Yeah, the behavior from `mouseenter` is kinda of buggy at least for me